miso-web/src/main/webapp/scripts/box_visualization.js

```diff
@@ -210,27 +177,56 @@ var BoxVisual = function() {
     jQuery(self.table).append(tBody);
     jQuery('#updateSelected, #removeSelected, #emptySelected').prop('disabled', true).addClass('disabled');
   };
-
-  self.getBoxPosition = function(row, col, tCell) {
-    // Override this method
-    return new BoxPosition({
-      row: row,
-      col: col,
-      cell: tCell,
-      boxItem: self.getBoxItem(row, col)
-    });
+  
+  self.selectRow = function(event) {
+    var row = event.data.row;
+    var items = [];
+    for (var col = 1; col <= self.size.cols; col++) {
+      items.push(self.position[row][col]);
+    }
+    toggleGroupSelection(event, items);
   };
-
-  self.click = function(row, col) {
-    var event = {
-      data: {
-        'row': row,
-        'col': col
+  
+  self.selectCol = function(event) {
+    var col = event.data.col;
+    var items = [];
+    for (var row = 1; row <= self.size.rows; row++) {
+      items.push(self.position[row][col]);
+    }
+    toggleGroupSelection(event, items);
+  };
+  
+  var toggleGroupSelection = function(event, items) {
```

Comment:
I know JS variables are function scoped, but it would be preferrable to declare `var toggleGroupSelection` before using it.